app/webchat_handlers.py

```diff
@@ -160,15 +164,8 @@ async def post(self, request):
         if not form_valid:
             logger.info('form submission error',
                         client_ip=request['client_ip'])
-            return {
-                'form_value_screen_name': data.get('screen_name'),
-                'form_value_country': data.get('country'),
-                'form_value_query': data.get('query'),
-                'display_region': display_region,
-                'page_title': page_title,
-                'locale': locale,
-                'page_url': View.gen_page_url(request)
-            }
+            raise HTTPFound(
+                request.app.router['WebChat:get'].url_for(display_region=display_region))
```

Comment:
If data error on POST why issue a redirect with a HTTP 302 and ask the browser to make another request. Why not just render the response with error messages in the reply?